webapp/channels/src/components/app_bar/app_bar.test.tsx

```diff
@@ -134,65 +48,141 @@ describe('components/app_bar/app_bar', () => {
         },
     ] as AppBinding[];
 
-    test('should match snapshot on mount', async () => {
-        const wrapper = mount(
+    const initialState = {
+        views: {
+            rhs: {
+                isSidebarOpen: true,
+                rhsState: 'plugin',
+                pluggableId: 'the_rhs_plugin_component',
+            },
+        },
+        plugins: {
+            components: {
+                AppBar: channelHeaderComponents,
+                RightHandSidebarComponent: rhsComponents,
+                Product: [],
+            } as {[componentName: string]: PluginComponent[]},
+        },
+        entities: {
+            apps: {
+                main: {
+                    bindings: channelHeaderAppBindings,
+                } as {bindings: AppBinding[]},
+                pluginEnabled: true,
+            },
+            general: {
+                config: {
+                    DisableAppBar: 'false',
+                    FeatureFlagAppsEnabled: 'true',
+                },
+            },
+            channels: {
+                currentChannelId: 'currentchannel',
+                channels: {
+                    currentchannel: TestHelper.getChannelMock({
+                        id: 'currentchannel',
+                    }),
+                },
+                myMembers: {
+                    currentchannel: TestHelper.getChannelMembershipMock({
+                        channel_id: 'currentchannel',
+                        user_id: 'user1',
+                    }),
+                },
+            },
+            teams: {
+                currentTeamId: 'currentteam',
+            },
+            users: {
+                currentUserId: 'user1',
+                profiles: {
+                    user1: TestHelper.getUserMock({
+                        roles: 'system_user',
+                    }),
+                },
+            },
+        },
+    };
+
+    test('should match snapshot on mount', () => {
+        const testState = initialState;
+        const {asFragment} = renderWithContext(
             <AppBar/>,
+            testState,
         );
 
-        expect(wrapper).toMatchSnapshot();
+        expect(asFragment()).toMatchSnapshot();
     });
 
-    test('should match snapshot on mount when App Bar is disabled', async () => {
-        mockState.entities.general.config.DisableAppBar = 'false';
+    test('should match snapshot on mount when App Bar is disabled', () => {
+        const testState = mergeObjects(initialState, {
+            entities: {
+                general: {
+                    config: {
+                        DisableAppbar: 'false',
+                    },
+                },
+            },
+        });
```

Comment:
It is fine to keep it like this, but I like the function approach a bit better.

If the initial state comes from a function, it is safe to edit the returned object. So this can be like...

```
const testState = getInitialState();
testState.entities.general.config.DisableAppbar = 'false';
```

In the next test would be:
```
const testState = getInitialState();
testState.entities.general.config.DisableAppbar = 'true';
testState.entities.general.config.FeatureFlagAppsEnabled = 'true';
testState.entities.general.config.EnableMarketpalce = 'true';
testState.entities.general.config.PluginsEnabled = 'true';
```

It stays more shallow, and I feel is easier to read what values we are modifying.

Nevertheless, this is just about taste, so feel free to ignore if you like it better this way.

Reply:
Yeah, that'd probably be a better way to do this all. We already have 75 or so files that do it with `mergeObjects` this way, so I wonder if there's a way to migrate those all at once.

I'm going to see if this is a good excuse for me to try out jscodeshift to see if I can do it everywhere. That'll be outside the scope of this PR though